Parser/pegen.c

```diff
@@ -836,6 +835,31 @@ _PyPegen_expect_token(Parser *p, int type)
     return t;
 }
 
+Token *
+_PyPegen_expect_forced_token(Parser *p, int type, const char* expected) {
+    if (!p->call_invalid_rules) {
+        return _PyPegen_expect_token(p, type);
+    }
```

Comment:
I added this because I liked to maintain the fact that custom syntax errors are not executed on the first run of the parser.

If you prefer, I can remove it.

Reply:
Not sure here. I'd probably leave it in in the first run as well, but it's up to you.

Reply:
Ok, I removed it in the last commit